contrib/win32/win32compat/misc.c

```diff
@@ -909,31 +918,44 @@ realpath(const char *path, char resolved[PATH_MAX])
 	return resolved;
 }
 
+/* This function is not thread safe. 
+* TODO - It uses static memory. Is this a good design?
+*/
 char*
-sanitized_path(const char *path)
+resolved_path(const char *input_path)
 {
-	if(!path) return NULL;
-
+	static char resolved_path[PATH_MAX] = {0,};
 	static char newPath[PATH_MAX] = { '\0', };
 	errno_t r = 0;
 
-	if (path[0] == '/' && path[1]) {
-		if (path[2] == ':') {
-			if (path[3] == '\0') { /* make "/x:" as "x:\\" */
-				if((r = strncpy_s(newPath, sizeof(newPath), path + 1, strlen(path) - 1)) != 0 ) {
-					debug3("memcpy_s failed with error: %d.", r);
-					return NULL;
-				}
-				newPath[2] = '\\';
-				newPath[3] = '\0';
-
-				return newPath;
+	if (!input_path) return NULL;
+
+	/* If filename contains __PROGRAMDATA__ then expand it to %programData% and return the resolved path */
+	if (NULL != strstr(input_path, PROGRAM_DATA)) {
```

Comment:
strstr() - this looks for first occurance anywhere in the string.. rather we are interested in knowing if path starts with __PROGRAMDATA__
Do
if (strlen(input_path) >= strlen(PROGRAM_DATA && memcmp(input_path, PROGRAM_DATA, strlen(PROGRAM_DATA) == 0